Tutorials/whatiswot/15-Protocol_Level/README.md

```diff
@@ -1,15 +1,193 @@
+
+``` json
+{
+  "properties": {
+    "coffeeBeansLeft": {
+      "title": "Remaining Coffee Beans",
+      "description": "Amount of remaining coffee beans in grams",
+      "type": "number",
+      "minimum": 0,
+      "maximum": 500,
+      "readOnly": true,
+      "observable": false,
+      "forms": [ ... ]
+    }
+  },
+  "actions": {
+    "brewCoffee": {
+      "title": "Brew Coffee",
+      "description": "Starts brewing a cup of coffee",
+      "input": {
+        "type": "object",
+        "properties": {
+          "size": {
+            "type": "string",
+            "enum": ["small", "medium", "large"]
+          }
+        }
+      },
+      "forms": [ ... ]
+    }
+  },
+  "events": {
+    "lowOnWater": { 
+        "title": "Low Water Level",
+        "forms": [ ... ]
+    }
+  }
+}
+
+```
+
+#### HTTP
+
+HTTP follows a request–response model and integrates naturally with existing web infrastructure. In WoT, HTTP bindings are typically used for reading and writing properties, invoking actions and accessing Thing metadata. HTTP is a good choice when devices are web-connected and when interoperability with existing web services is important.
+
+```json
+"properties": {
+  "coffeeBeansLeft": {
+    "title": "Remaining Coffee Beans",
+    "description": "Amount of remaining coffee beans in grams",
+    "type": "number",
+    "readOnly": true,
+    "forms": [
+      {
+        "href": "https://coffee.example.com/properties/coffeeBeansLeft",
+        "op": "readproperty",
+        "contentType": "application/json"
+      }
+    ]
+  }
+}
+```
+
+Here, the form tells the Consumer that it can read the property `coffeeBeansLeft` using HTTP and expects a JSON response. Because this is a readproperty operation, the Consumer knows to perform an HTTP GET request.
+
+#### CoAP
+
+CoAP is a lightweight protocol designed for constrained devices and low-power networks. It provides a similar interaction model to HTTP, but with much lower overhead, making it suitable for embedded systems and IoT deployments with limited resources.
+
+Let's use CoAP to invoke the `brewCoffee` action:
+
+```json
+"actions": {
+  "brewCoffee": {
+    "title": "Brew Coffee",
+    "description": "Starts brewing a cup of coffee",
+    "input": {
+      "type": "object",
+      "properties": {
+        "size": {
+          "type": "string",
+          "enum": ["small", "medium", "large"]
+        }
+      }
+    },
+    "forms": [
+      {
+        "href": "coap://coffee.example.com/actions/brewCoffee",
+        "op": "invokeaction",
+        "contentType": "application/json"
+      }
+    ]
+  }
+}
+```
+
+This time, the form uses a `coap://` URI to indicate CoAP instead of HTML. From the Consumer's perspective, invoking the action works the same as with HTTP - the only difference is the underlying protocol, which is optimized for constrained devices.
```

Comment:
```suggestion
This time, the form uses a `coap://` URI to indicate CoAP instead of HTTP. From the Consumer's perspective, invoking the action works the same as with HTTP - the only difference is the underlying protocol, which is optimized for constrained devices.
```